File: pyround1/231.2-的幂.py
#
# @lc app=leetcode.cn id=231 lang=python
#
# [231] 2的幂
#

# @lc code=start
class Solution(object):
    def isPowerOfTwo(self, n):
        """
        :type n: int
        :rtype: bool
        """
        while n >1:     
            if n %2 == 0:
                n /= 2
            else:
                return False
        return n == 1
# @lc code=end

File: 231.2-的幂.cpp
/*
 * @lc app=leetcode.cn id=231 lang=cpp
 *
 * [231] 2的幂
 */

// @lc code=start
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if(n <= 0)
            return false;
        if(n == 1)
            return true;
        while(n > 1){
            if(n %2 != 0)
                return false;
            n /= 2;
        }
        return true;
    }
};
// @lc code=end

